File: DS/api/java/GetIPSRules.java
import.com.trendmicro.deepsecurity.ApiClient;

/* class will get Intrusion Prevention Rules from computers by Deep Security
 * Manager using APIs
 */

 public class GetIPSRules {

    public static void main(String[] args) {
    // Setup the connection to Deep Security Manager
    // You need to enter your DSM URL
    ApiClient defaultClient = Configuration.getDefaultApi();
    // BasePath needs to be https://<your DSM FQDN or IP>:119/api
    defaultClient.setBasePath("<Your Hostname>:4119/api");

    // Authentication
    // You need to enter your API Key
    ApiKeyAuth DefaultAuthentication = (ApiKeyAuth) defaultClient.getAuthenication("DefaultAuthentication");
    DefaultAuthentication.setApiKey ("<Your API Key>");
    try {
       defaultClient.trustAllCertificates(false);
    } catch (Exception e) {
       System.err.println ("An exception occured when calling ApiClient.trustAllCertificates");
       e.printStackTrace();  
    }

    // Initialization
    // Get the information from IPS API
    IntrusionPreventionRulesAPI instance = new IntrusionPreventionRulesApi();
    
    String apiVersion = "v1";
    try {
       // get the list of Intrusion Prevention Rules from DSM
       // and display the data
       IntrusionPreventionRules result = instance.listIntrusionPreventionRules(apiVersion);
       system.out.println(result);
    } catch (ApiException e) {
       System.err.println("An exception occured when calling IntrusionPreventionRulesApi.listComputers");
       e.printStackTrace();
    } 
    }
 }

File: DS/api/python/get_ips_rules.py
from __future__ import print_function
import sys, warnings
import deepsecurity
from deepsecurity.rest import ApiException
from print import pprint

# Setup
if not sys.warnoptions:
   warnings.simplefilter("ignore")
configuration = deepsecurity.Configuration()
# host must be set to 
configuration.host = '<Your Hostname or IP>'

# Authentication
# The API key must be set to
configuration.api_key['api-secret-key'] = '<Your API Key>'

# Initialization
# Set any required values
api_instance = deepsecurity.IntrusionPreventionRulesApi (deepsecurity.ApiClient(configuration))
api_version = 'v1'

try: 
   api_response = api_instance.list_intrusion_prevention_rules(api_version)
   pprint(api_response)
except ApiException as e:
   print ("An exception occurred when calling IntrusionPreventionRulesApi.list_intrusion_prevention_rules: %s\n" % e)
